core/trino-main/src/main/java/io/trino/operator/output/PagePartitioner.java

```diff
@@ -146,11 +148,51 @@ public void partitionPage(Page page, OperatorContext operatorContext)
         else {
             partitionPageByColumn(page);
         }
-        long positionsAppendersSizeAfter = getPositionsAppendersSizeInBytes();
-        flushPositionsAppenders(false);
+        long outputSizeInBytes = flushPositionsAppenders(false);
         updateMemoryUsage();
 
-        operatorContext.recordOutput(positionsAppendersSizeAfter - positionsAppendersSizeBefore, outputPositionCount);
+        recordOutputStats(operatorContext, outputSizeInBytes, outputPositionCount);
+    }
+
+    private void recordOutputStats(OperatorContext operatorContext, long outputSizeInBytes, int outputPositions)
+    {
+        // The outputSizeInBytes already reported during release
+        if (outputSizeReportedBeforeRelease > 0) {
+            long consumeFromPreviouslyReported = min(outputSizeReportedBeforeRelease, outputSizeInBytes);
+            outputSizeReportedBeforeRelease -= consumeFromPreviouslyReported;
```

Comment:
Why would `outputSizeReportedBeforeRelease` go down?

Reply:
Because `outputSizeReportedBeforeRelease` represents output bytes that have already been reported in advance of the page being flushed as a result of the partitioner being released by a previous operator. In this logic we're "consuming" from the eagerly reported size since a page flush has occurred.